include/hpcombi/hpcombi.hpp

```diff
@@ -30,3 +34,73 @@
 #include "vect_generic.hpp"
 
 #endif  // HPCOMBI_HPCOMBI_HPP_
+
+/*! \mainpage HPCombi
+
+\section readme_sec Readme
+
+You might want to have a look at [the Readme in the sources](https://github.com/libsemigroups/HPCombi/blob/main/README.md).
+
+\section sec_philo Philosophy
+This library provides high performance computations in combinatorics (hence its name).
+In practice we observe large speedups in several enumeration problems.
+
+The main idea of the library is a way to encode data as a small sequence of small integers,
+that can be handled efficiently by a creative use of vector instructions.
+For example, on the current x86 machines, small permutations (N ≤ 16) are very well handled.
+Indeed thanks to machine instructions such as PSHUFB (Packed SHUFfle Bytes),
+applying a permutation on a vector only takes a few CPU cycles.
+
+Further ideas are:
+- Vectorization (MMX, SSE, AVX instructions sets) and careful memory alignment,
+- Careful memory management: avoid all dynamic allocation during the computation,
+- Avoid all unnecessary copies (it is often needed to rewrite the containers),
+- Due to combinatorial explosion, sets often don’t fit in memory or disk and are enumerated on the fly.
+
+Here are some examples,
+the speedup is in comparison to an implementation without vector instructions:
+
+Operation |   Speedup
+----------|-----------
+Inverting a permutation | 1.28
+Sorting a list of bytes | 21.3
+Number of cycles of a permutation |  41.5
+Number of inversions of a permutation  | 9.39
+Cycle type of a permutation | 8.94
+
+
+
+\section sec_tips Tips to the user
+
+Note that memory access can become a problem. It you store many things, most of the time will be spent in fetching from RAM, not computing.
+Data structure should preserve locality. You might want to compute some stats on data structure usage and write custom ones.
```

Comment:
```suggestion
Data structures should preserve locality. You might want to compute some stats on data structure usage and write custom ones.
```
Fix grammar, also: these two sentences are probably too general to convey much information: what data structures? what sort of stats? custom data structures for what?

Reply:
I added more details.